docs/TrafficSplitting.md

```diff
@@ -0,0 +1,163 @@
+
+*Note*:
+- `--untag-revision` flag's action takes first precedence, meaning if `--untag-revision` and `--tag-revision` flags specified together for revisions, all the mentioned tags will be unassigned from revisions first followed by tagging them.
+- Using `--untag-revision` with `--tag-revision` can be used to *update* tag for a particular Revision, for eg: to update Tag from `current` to `stable` for a Revision `echo-v3`, do `kn service update svc --untag-revision current --tag-revision echo-v3:stable`
+- If a Revision is untagged and it has 0% traffic split, its removed from the traffic block entirely.
+
+## Flag operation precedence
+- Mentioned 5 flags for traffic splitting and tagging operations can be specified together in a single `kn service update` command, thus `kn` defines the precedence of these flags in which operations are evaluated.
+- The order of the flags specified in the command are **not** taken into account.
+- Enlisting the precedence of flag(s) as they are evaluated by `kn`:
+
+1. `--untag-revision` : First, all the referenced tags by this flag are removed from the traffic block
+2. `--tag-revision` or `--tag-latest` : Second, all the revisions are tagged as specified in the traffic block
+3. `--traffic` or `--traffic-latest`: Third, all the referenced revisions (or tags) are assigned given portion of traffic split
+
+Thus, if a user wants to update tag for revision `echo-v2` from `new` to `old` with `50%` traffic while adjusting rest to latest revision, will do
+```
+kn service update svc --untag-revision new --tag-revision echo-v2:old --traffic old:50 --traffic-latest 50
+```
+
+## Summary
+Following table shows quick summary for traffic splitting flags, their value formats, operation to perform and *Repetition* column denotes if repeating the flag is allowed in a `kn service update` command.
+
+
+Flag | Value(s) | Operation | Repetition
+:--- | :---: | :--- | :---:
+`--traffic` | `RevisionName:Percent` | Give `Percent` traffic to `RevisionName` | :heavy_check_mark:
+`--traffic` | `Tag:Percent` | Give `Percent` traffic to the Revision having `Tag` | :heavy_check_mark:
+`--traffic-latest` | `Percent` | Give `Percent` traffic to the latest ready Revision | :heavy_multiplication_x:
+`--tag-revision` | `RevisionName:Tag` | Give `Tag` to `RevisionName` | :heavy_check_mark:
+`--tag-latest` | `Tag` | Give `Tag` to the latest ready Revision | :heavy_multiplication_x:
+`--untag-revision` | `Tag` | Remove `Tag` from Revision | :heavy_check_mark:
+
+
+## Examples
+1. Tag revisions `echo-v1` and `echo-v2` as `stable` and `staging` respectively:
+```
+kn service update svc --tag-revision echo-v1:stable --tag-revision echo-v2:staging
+```
+
+2. Ramp up/down revision `echo-v3` to `20%`, adjusting other traffic to accommodate:
+```
+kn service update svc --traffic echo-v3:20 --traffic echo-v2:80
+```
+
+3. Give revision `echo-v3` the tag `candidate`, without otherwise changing any traffic split:
```

Comment:
If the `echo-v3` have an existing tag, this will create a new revision, right? If true, I think we should highlight it to users.

Reply:
It won't, any modification in traffic block of service doesn't create a new revision.